File: kvdom-dom/src/wasm32Main/resources/bindings/Element.js
konan.libraries.push({
    kvdom_Element_setAttribute: function (arena, obj, keyPtr, keyLen, valuePtr, valueLen) {
        let key = toUTF16String(keyPtr, keyLen);
        let value = toUTF16String(valuePtr, valueLen);
        kotlinObject(arena, obj).setAttribute(key, value);
    },
    kvdom_Element_removeAttribute: function (arena, obj, keyPtr, keyLen) {
        let key = toUTF16String(keyPtr, keyLen);
        kotlinObject(arena, obj).removeAttribute(key);
    }
});

File: kvdom/kvdom-dom/src/wasm32Main/kotlin/lt/petuska/kvdom/dom/Element.kt
package lt.petuska.kvdom.dom

import kotlinx.wasm.jsinterop.Arena
import kotlinx.wasm.jsinterop.Object
import kotlinx.wasm.jsinterop.Pointer
import kotlinx.wasm.jsinterop.stringLengthBytes
import kotlinx.wasm.jsinterop.stringPointer

public actual abstract class Element(arena: Arena, index: Object) : Node(arena, index) {
  public actual fun removeAttribute(qualifiedName: String) {
    js_removeAttribute(
      arena, index,
      stringPointer(qualifiedName), stringLengthBytes(qualifiedName),
    )
  }

  public actual fun setAttribute(qualifiedName: String, value: String) {
    js_setAttribute(
      arena, index,
      stringPointer(qualifiedName), stringLengthBytes(qualifiedName),
      stringPointer(value), stringLengthBytes(value),
    )
  }
}

@SymbolName("kvdom_Element_setAttribute")
private external fun js_setAttribute(
  arena: Arena,
  index: Object,
  keyPtr: Pointer,
  keyLen: Int,
  valuePtr: Pointer,
  valueLen: Int,
)

@SymbolName("kvdom_Element_removeAttribute")
private external fun js_removeAttribute(
  arena: Arena,
  index: Object,
  keyPtr: Pointer,
  keyLen: Int,
)
